File: app/storage.py
import os
from pathlib import Path


def _base_dir() -> Path:
    return Path(os.environ.get("DATA_DIR", "/data"))


def get_upload_path(job_id: str) -> Path:
    path = _base_dir() / "uploads" / job_id
    path.mkdir(parents=True, exist_ok=True)
    return path / "input.pdf"


def get_output_path(job_id: str) -> Path:
    path = _base_dir() / "outputs" / job_id
    path.mkdir(parents=True, exist_ok=True)
    return path / "output.mp3"


def get_batch_dir(job_id: str) -> Path:
    path = _base_dir() / "outputs" / job_id / "batches"
    path.mkdir(parents=True, exist_ok=True)
    return path

File: service/server/services/storage.ts
import { mkdir } from 'node:fs/promises'
import { join } from 'node:path'

function dataDir(): string {
  return process.env.DATA_DIR ?? '/data'
}

export function getUploadPath(jobId: string): string {
  return join(dataDir(), 'uploads', jobId, 'input.pdf')
}

export function getOutputPath(jobId: string): string {
  return join(dataDir(), 'outputs', jobId, 'output.mp3')
}

export async function ensureUploadDir(jobId: string): Promise<string> {
  const dir = join(dataDir(), 'uploads', jobId)
  await mkdir(dir, { recursive: true })
  return join(dir, 'input.pdf')
}
